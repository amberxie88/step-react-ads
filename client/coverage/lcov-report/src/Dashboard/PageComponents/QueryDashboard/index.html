
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/Dashboard/PageComponents/QueryDashboard</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> src/Dashboard/PageComponents/QueryDashboard</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">77.03% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>57/74</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">46.67% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>7/15</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">77.27% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>17/22</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.71% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>56/73</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Queries.js"><a href="Queries.js.html">Queries.js</a></td>
	<td data-value="93.55" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 93%"></div><div class="cover-empty" style="width: 7%"></div></div>
	</td>
	<td data-value="93.55" class="pct high">93.55%</td>
	<td data-value="31" class="abs high">29/31</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="9" class="abs high">8/9</td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="30" class="abs high">28/30</td>
	</tr>

<tr>
	<td class="file low" data-value="QueryAccount.js"><a href="QueryAccount.js.html">QueryAccount.js</a></td>
	<td data-value="47.37" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 47%"></div><div class="cover-empty" style="width: 53%"></div></div>
	</td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	</tr>

<tr>
	<td class="file high" data-value="QueryDashboard.js"><a href="QueryDashboard.js.html">QueryDashboard.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file medium" data-value="QueryResults.js"><a href="QueryResults.js.html">QueryResults.js</a></td>
	<td data-value="76.19" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 76%"></div><div class="cover-empty" style="width: 24%"></div></div>
	</td>
	<td data-value="76.19" class="pct medium">76.19%</td>
	<td data-value="21" class="abs medium">16/21</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="76.19" class="pct medium">76.19%</td>
	<td data-value="21" class="abs medium">16/21</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Jul 23 2020 17:58:10 GMT-0400 (Eastern Daylight Time)
            </div>
        </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    