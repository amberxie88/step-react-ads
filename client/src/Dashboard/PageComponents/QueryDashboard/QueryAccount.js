/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import axios from 'axios';
import Title from '../../Utilities/Title';
import Typography from '@material-ui/core/Typography';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';

class QueryAccount extends React.Component {
  constructor(props) {
    super(props);
    this.getClient = this.getClient.bind(this);
    this.state = {
      selectedClient: [],
      status: 'none selected',
    };
  }

  async getClient() {
    this.setState({ status: 'loading' });
    const { data } = await axios.get('/client');
    console.log(data);
    if (data.loginId) {
      this.setState({
        selectedClient: data,
        status: 'loaded',
      });
    } else {
      this.setState({ status: 'none selected' });
    }
  }

  pickContentToDisplay() {
    const selectedClient = this.state.selectedClient;
    switch (this.state.status) {
      case 'none selected':
        return (
          <Typography variant="overline">
            Select a client on the login page.
          </Typography>
        );
      case 'loading':
        return <Typography variant="overline">Loading . . .</Typography>;
      case 'loaded':
        return (
          <TableBody>
            <TableRow>
              <TableCell key={selectedClient.loginId}>
                {selectedClient.loginId}
              </TableCell>
              <TableCell key={selectedClient.customerId}>
                {selectedClient.customerId}
              </TableCell>
              <TableCell key={selectedClient.name}>
                {selectedClient.name}
              </TableCell>
            </TableRow>
          </TableBody>
        );
      default:
        return (
          <Typography variant="overline">
            Select a client on the login page.
          </Typography>
        );
    }
  }

  componentDidMount() {
    this.getClient();
  }

  render() {
    const selectedClient = this.state.selectedClient;
    console.log(selectedClient);
    return (
      <React.Fragment>
        <Title>Selected Client Account to Query</Title>
        <Table size="small">
          <TableHead>
            <TableRow>
              <TableCell>Customer ID</TableCell>
              <TableCell>Client Account ID</TableCell>
              <TableCell>Client Account Name</TableCell>
            </TableRow>
          </TableHead>
          {this.pickContentToDisplay()}
        </Table>
      </React.Fragment>
    );
  }
}

export default QueryAccount;
