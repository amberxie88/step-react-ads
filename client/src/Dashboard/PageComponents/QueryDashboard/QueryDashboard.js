import React from 'react';
import clsx from 'clsx';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Query from './Queries';
import { makeStyles } from '@material-ui/core/styles';

import { stylesSettings } from '../../Utilities/Styles';
const Styles = makeStyles(stylesSettings);

export default function Queries(props) {
  const classes = Styles();
  return (
    <React.Fragment>
      <Grid container spacing={3}>
        {/* CampaignData */}
        <Grid item xs={12}>
          <Paper className={classes.paper}>
            <Query />
          </Paper>
        </Grid>
      </Grid>
    </React.Fragment>
  );
}
