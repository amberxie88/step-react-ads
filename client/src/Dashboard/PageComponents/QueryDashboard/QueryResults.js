import React from 'react';
import Title from '../../Utilities/Title';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TablePagination from '@material-ui/core/TablePagination';

export default function QueryResults(props) {
  const { rows, fields } = props;
  const [rowsPerPage, setRowsPerPage] = React.useState(10);
  const [page, setPage] = React.useState(0);

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    // If the selection is [A]ll the rows.
    if (event.target.value.toString().charAt(0) === 'A') {
      event.target.value = rows.length;
    }
    setRowsPerPage(+event.target.value);
    setPage(0);
  };
  return (
    <React.Fragment>
      <Title>Query Results</Title>
      <Results
        rows={rows}
        fields={fields}
        rowsPerPage={rowsPerPage}
        page={page}
        handleChangePage={handleChangePage}
        handleChangeRowsPerPage={handleChangeRowsPerPage}
      />
    </React.Fragment>
  );
}

function Results(props) {
  const rows = props.rows;
  const fields = props.fields;
  const rowsPerPage = props.rowsPerPage;
  const page = props.page;
  const handleChangePage = props.handleChangePage;
  const handleChangeRowsPerPage = props.handleChangeRowsPerPage;
  //console.log(rows);
  //console.log(fields);
  return (
    <div>
      <Table size="small">
        <TableHead>
          <TableRow>
            {fields.map((key, index) => (
              <TableCell key={'col' + index}>{key}</TableCell>
            ))}
          </TableRow>
        </TableHead>
        <TableBody>
          {rows
            .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
            .map((row) => (
              <TableRow key={'row' + row.id}>
                {fields.map((key) => (
                  <TableCell key={key}>{row[key]}</TableCell>
                ))}
              </TableRow>
            ))}
        </TableBody>
      </Table>
      <TablePagination
        rowsPerPageOptions={[
          5,
          10,
          25,
          'All ' + rows.length.toString() + ' Rows',
        ]}
        component="div"
        count={rows.length}
        rowsPerPage={rowsPerPage}
        page={page}
        onChangePage={handleChangePage}
        onChangeRowsPerPage={handleChangeRowsPerPage}
      />
    </div>
  );
}
