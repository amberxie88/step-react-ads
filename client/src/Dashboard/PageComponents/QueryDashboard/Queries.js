/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import axios from 'axios';
import * as HttpStatus from 'http-status-codes';
import { makeStyles } from '@material-ui/core/styles';
import QueryResults from './QueryResults';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import Title from '../../Utilities/Title';

const useStyles = makeStyles((theme) => ({
  root: {
    margin: theme.spacing(1),
  },
}));

//JSON data parsing here
function addKey(index, responseObj) {
  responseObj.id = index;
  return responseObj;
}

function parseRows(response) {
  const responseRows = [];
  let i;
  for (i = 0; i < response.length; i++) {
    responseRows.push(addKey(i, response[i]));
  }
  return responseRows;
}

class Query extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    this.handleQuery = this.handleQuery.bind(this);
    this.state = {
      value: '',
      rows: [],
      fields: [],
      rowsPerPage: 5,
      status: 'inputRequired',
      errorMessage: '',
    };
  }

  async handleQuery() {
    const query = this.state.value;
    this.setState({ status: 'loading' });
    try {
      const { data } = await axios.post(
        '/campaign',
        new URLSearchParams({ query }),
      );
      if (data.meta.status !== HttpStatus.OK.toString()) {
        throw new Error(data.meta.message);
      } else {
        this.setState({
          rows: parseRows(data.response),
          fields: data.fieldmask,
          status: 'loaded',
        });
      }
    } catch (err) {
      console.log(err.message);
      this.setState({ status: 'error', errorMessage: err.message });
    }
  }

  pickContentToDisplay() {
    switch (this.state.status) {
      case 'inputRequired':
        return <Title> Enter a query to see the results </Title>;
      case 'loading':
        return <Title> Loading ... </Title>;
      case 'loaded':
        return (
          <QueryResults
            rows={this.state.rows}
            fields={this.state.fields}
            rowsPerPage={this.state.rowsPerPage}
          />
        );
      case 'error':
        return (
          <Title>
            {"Something Went Wrong. Here's the Error Message: " +
              this.state.errorMessage}
          </Title>
        );
      default:
        return <Title> Something Went Wrong</Title>;
    }
  }

  handleChange(event) {
    this.setState({ value: event.target.value });
  }

  render() {
    // const rows = this.state.rows;
    // const fields = this.state.fields;
    return (
      <React.Fragment>
        <Title>Query Here</Title>
        <TextField
          value={this.state.value}
          onChange={this.handleChange}
          id="standard-full-width"
          label="Enter Query"
          style={{ margin: 8 }}
          placeholder="Placeholder"
          helperText="make sure to select an account!"
          fullWidth
          margin="normal"
          InputLabelProps={{
            shrink: true,
          }}
        />
        <SubmitButton onClick={this.handleQuery} />
        {this.pickContentToDisplay()}
      </React.Fragment>
    );
  }
}

function SubmitButton(props) {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Button variant="outlined" onClick={props.onClick}>
        Submit
      </Button>
    </div>
  );
}

export default Query;
