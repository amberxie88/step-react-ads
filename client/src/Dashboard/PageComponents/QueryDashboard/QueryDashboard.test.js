import React from 'react';
React.useLayoutEffect = React.useEffect; // Necessary to supress warning during tests. The tests use server side rendering which doesn't support useLayoutEffect.
import QueryDashboard from './QueryDashboard';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('QueryDashboard Unit Testing', () => {
  it('renders correct HTML for Query Dashboard', async () => {
    const component = shallow(<QueryDashboard />);
    const componentHTML = component.html();
    expect(componentHTML).toMatchSnapshot();
  });
});
