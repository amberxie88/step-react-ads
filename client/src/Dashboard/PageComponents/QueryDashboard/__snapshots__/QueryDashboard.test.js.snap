// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryDashboard Unit Testing renders correct HTML for Query Dashboard 1`] = `"<div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiPaper-root makeStyles-paper-14 MuiPaper-elevation1 MuiPaper-rounded\\"><h2 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom\\">Selected Client Account to Query</h2><table class=\\"MuiTable-root\\"><thead class=\\"MuiTableHead-root\\"><tr class=\\"MuiTableRow-root MuiTableRow-head\\"><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall\\" scope=\\"col\\">Customer ID</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall\\" scope=\\"col\\">Client Account</th></tr></thead><span class=\\"MuiTypography-root MuiTypography-overline\\">Select a client on the login page.</span></table></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiPaper-root makeStyles-paper-14 MuiPaper-elevation1 MuiPaper-rounded\\"><h2 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom\\">Query Here</h2><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\" style=\\"margin:8px\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink\\" data-shrink=\\"true\\" for=\\"standard-full-width\\" id=\\"standard-full-width-label\\">Enter Query</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-describedby=\\"standard-full-width-helper-text\\" id=\\"standard-full-width\\" placeholder=\\"Placeholder\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div><p class=\\"MuiFormHelperText-root\\" id=\\"standard-full-width-helper-text\\">make sure to select an account!</p></div><div class=\\"makeStyles-root-16\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Submit</span></button></div><h2 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom\\"> Enter a query to see the results </h2></div></div></div>"`;
